docs/site/migration/LB3-vs-LB4-booting.md

```diff
@@ -0,0 +1,153 @@
+which performs the following tasks:
+
+#### 1. Configuration of application settings
+
+It determines the execution environment, the application host, the application
+port, the application root directory and other artifact directories.
+
+Then it configures the LoopBack Application object from the `config.json` file
+in the application root directory.
+
+#### 2. Configuration of datasources
+
+Datasources are determined from application options or datasource files. They
+are then attached to the LoopBack Application object.
+
+#### 3. Definition of models
+
+Models are determined from application options or model files,
+[mixins](https://loopback.io/doc/en/lb3/Defining-mixins.html) are applied,
+and they are attached to their respective datasources. They are then attached
+to the LoopBack Application object.
+
+#### 4. Setting up middleware
+
+Middleware from the middleware configuration file are resolved and added to the
+LoopBack 3 middleware chain.
+
+#### 5. Setting up components
+
+Components from the component configuration file are resolved and are added to
+the LoopBack 3 middleware chain.
+
+#### 6. Running boot scripts
+
+Boot scripts are resolved from the `server/boot` directory and run.
+
+### In LoopBack 4
+
+In LoopBack 4, the booting tasks are shared between the `@loopback/core`'s
+[Application](./apidocs/apidocs.core.application.md) class
+and the
+[@loopback/boot](./apidocs/apidocs.boot.html) package. Since your application
+will extend `BootMixin` and `Application`, their separation will not be
+apparent to you, and understanding of how they work together as one is not
+mandatory for developing LoopBack 4 projects.
+
+The following are the list of tasks that are performed in the LoopBack 3
+booting process, and their equivalent in LoopBack 4.
+
+{% include tip.html content="For all the details about booting in LoopBack 4,
+refer to the
+[booting documentation](./Booting-an-Application.md)." %}
+
+### 1. Configuration of application settings
+
+There is no `config.json` in LoopBack 4. The application options are passed in
+the constructor of the
+[Application](./apidocs/apidocs.core.application.md) class. Read more about
+application configuration
+[here](./Application.md#configuring-your-application).
+
+### 2. Configuration of datasources
+
+Datasources in LoopBack 3 are defined in a single `datasources.json` file. In
+LoopBack 4, datasources are defined as TypeScript classes that come with an
+associated JSON configuration file. They all reside in the `src/datasources`
+directory.
+
+{% include note.html content="We are moving away from JSON files for datasource
+configuration. Refer to
+[loopback-next#5000](https://github.com/strongloop/loopback-next/pull/5000)
+for the progress." %}
+
+`BootMixin` resolves the datasources, and attaches them to the application.
+Read more about datasource booter
+[here](./Booting-an-Application.md#controller-booter).
+
+Read more about LoopBack 4 datasource [here](./DataSources.md).
+
```

Comment:
Should we add a link to https://loopback.io/doc/en/lb4/migration-datasources.html too?